engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -1048,16 +1058,33 @@ namespace dmGameSystem
                             &attribute_infos);
 
                 dmVMath::Matrix4 normal_matrix = dmRender::GetNormalMatrix(render_context, instance_render_item->m_World);
-                dmRig::WriteVertexAttributeParams params = {};
-                FillWriteVertexAttributeParams(instance_render_item->m_Mesh,
-                    dmGraphics::VERTEX_STEP_FUNCTION_INSTANCE,
+
+            #define UNPACK_ATTRIBUTE_PTR(name) \
+                (instance_render_item->m_Mesh->name.m_Count ? instance_render_item->m_Mesh->name.m_Data : 0)
+
+                const float* uv_channels[] = {
+                    UNPACK_ATTRIBUTE_PTR(m_Texcoord0),
+                    UNPACK_ATTRIBUTE_PTR(m_Texcoord1),
+                };
+
+                uint32_t uv_channels_count = (uv_channels[0] ? 1 : 0) + (uv_channels[1] ? 1 : 0);
+
+                dmGraphics::WriteAttributeParams params = {};
+                dmRig::SetMeshWriteAttributeParams(&params,
                     &attribute_infos,
+                    dmGraphics::VERTEX_STEP_FUNCTION_INSTANCE,
                     &instance_render_item->m_World,
                     &normal_matrix,
-                    &params);
+                    0, // World space positions are not supported by local space materials
```

Comment:
We circumvrent this later on by taking the "world coordinate" route in the batch function. If we have to produce world space positions for local spaced materials we would essentially need to do a third "mixed" approach that can update dynamic streams with static content (similar to the instance buffer). It's a lot of work to make that work, but maybe we can solve that later? The current setup will work, but not optimally I guess.